llama_index/indices/postprocessor/node_recency.py

```diff
@@ -130,59 +164,115 @@ def postprocess_nodes(
         if query_bundle is None:
             raise ValueError("Missing query bundle in extra info.")
 
-        # query_bundle = cast(QueryBundle, metadata["query_bundle"])
-        # infer_recency_prompt = SimpleInputPrompt(self.infer_recency_tmpl)
-        # raw_pred = self.service_context.llm_predictor.predict(
-        #     prompt=infer_recency_prompt,
-        #     query_str=query_bundle.query_str,
-        # )
-        # pred = parse_recency_pred(raw_pred)
-        # # if no need to use recency post-processor, return nodes as is
-        # if not pred:
-        #     return nodes
-
         # sort nodes by date
         node_dates = pd.to_datetime(
             [node.node.metadata[self.date_key] for node in nodes]
         )
         sorted_node_idxs = np.flip(node_dates.argsort())
         sorted_nodes: List[NodeWithScore] = [nodes[idx] for idx in sorted_node_idxs]
 
-        # get embeddings for each node
-        embed_model = self.service_context.embed_model
-        for node in sorted_nodes:
-            embed_model.queue_text_for_embedding(
-                node.node.node_id,
-                node.node.get_content(metadata_mode=MetadataMode.EMBED),
+        if self.embedding_filter_level == "documents":
+            sorted_doc_ids = list(
+                dict.fromkeys(
+                    [
+                        node.node.relationships[NodeRelationship.SOURCE].node_id
+                        for node in sorted_nodes
+                        if node.node.relationships[NodeRelationship.SOURCE] is not None
+                        and node.node.relationships[NodeRelationship.SOURCE].node_id
+                        is not None
+                    ]
+                )
             )
 
-        _, text_embeddings = embed_model.get_queued_text_embeddings()
-        node_ids_to_skip: Set[str] = set()
-        for idx, node in enumerate(sorted_nodes):
-            if node.node.node_id in node_ids_to_skip:
-                continue
-            # get query embedding for the "query" node
-            # NOTE: not the same as the text embedding because
-            # we want to optimize for retrieval results
-
-            query_text = self.query_embedding_tmpl.format(
-                context_str=node.node.get_content(metadata_mode=MetadataMode.EMBED),
-            )
-            query_embedding = embed_model.get_query_embedding(query_text)
-
-            for idx2 in range(idx + 1, len(sorted_nodes)):
-                if sorted_nodes[idx2].node.node_id in node_ids_to_skip:
+            sorted_node_ids_for_docs: List[List[str]] = []
+            for doc_id in sorted_doc_ids:
+                doc_info = self.storage_context.docstore.get_ref_doc_info(doc_id)
+                if doc_info is not None and doc_info.node_ids is not None:
+                    sorted_node_ids_for_docs.append(doc_info.node_ids)
+                else:
+                    sorted_node_ids_for_docs.append([])
```

Comment:
you can probably pull this out to a utils function: given node, retrieve all nodes of its source document.